### PR TITLE
Internal: Get information on user subscription [ED-14095]

### PR DESCRIPTION
**Pull request by hein-obox**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: modules/home/classes/transformations-manager.php [hunk 1]
```diff
@@ -33,10 +38,26 @@ public function run_transformations(): array {
...
 	}
 
-	private function run_transformation( $id ): array {
-		$class_name = '\\Elementor\\Modules\\Home\\Transformations\\' . $id;
-		$transformer = new $class_name( $this->home_screen_data, $this->wordpress_adapter );
+	private function run_transformation( $transformation_id ): array {
+		return $this->transformation_classes[ $transformation_id ]->transform();
+	}
+
+	private function set_transformation_classes(): array {
```
Comment (mykytamurzin) [inline modules/home/classes/transformations-manager.php:45]: maybe `get_transformation_classes` ?

File: tests/phpunit/elementor/modules/home/transformations/test-filter-plugins.php [hunk 2]
```diff
@@ -13,7 +13,11 @@ class Test_Filter_Plugins extends Elementor_Test_Base {
 	public function test_filter_plugins() {
```
Comment (mykytamurzin) [inline tests/phpunit/elementor/modules/home/transformations/test-filter-plugins.php:13]: should be `test_transform`

File: tests/phpunit/elementor/modules/home/transformations/test-remove-sidebar-upgrade-for-pro-users.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
+<?php
+namespace Elementor\Tests\Phpunit\Elementor\Modules\Home\Transformations;
+
+use Elementor\Modules\Home\Transformations\Remove_Sidebar_Upgrade_For_Pro_Users;
+use ElementorEditorTesting\Elementor_Test_Base;
+
+class Test_Remove_Sidebar_Upgrade_For_Pro_Users extends Elementor_Test_Base {
+
+	public function test_core_plugin() {
```
Comment (mykytamurzin) [inline tests/phpunit/elementor/modules/home/transformations/test-remove-sidebar-upgrade-for-pro-users.php:9]: name should be `test_{test_function_name}__{scenario}`

Comment (mykytamurzin) [reply tests/phpunit/elementor/modules/home/transformations/test-remove-sidebar-upgrade-for-pro-users.php:9]: so it should be something like `test_transform__core(has_core)`

File: modules/home/classes/transformations-manager.php (hunk 1 above)
Comment (hein-obox) [reply modules/home/classes/transformations-manager.php]: Updated.

File: tests/phpunit/elementor/modules/home/transformations/test-filter-plugins.php (hunk 2 above)
Comment (hein-obox) [reply tests/phpunit/elementor/modules/home/transformations/test-filter-plugins.php]: Updated.

File: tests/phpunit/elementor/modules/home/transformations/test-remove-sidebar-upgrade-for-pro-users.php (hunk 3 above)
Comment (hein-obox) [reply tests/phpunit/elementor/modules/home/transformations/test-remove-sidebar-upgrade-for-pro-users.php]: Updated.

**Review by murzin-nikita — DISMISSED** [review]

File: tests/phpunit/elementor/modules/home/transformations/test-remove-sidebar-upgrade-for-pro-users.php [hunk 4]
```diff
@@ -0,0 +1,65 @@
+<?php
+namespace Elementor\Tests\Phpunit\Elementor\Modules\Home\Transformations;
+
+use Elementor\Modules\Home\Transformations\Remove_Sidebar_Upgrade_For_Pro_Users;
+use ElementorEditorTesting\Elementor_Test_Base;
+
+class Test_Remove_Sidebar_Upgrade_For_Pro_Users extends Elementor_Test_Base {
```
Comment (louiswol94) [inline tests/phpunit/elementor/modules/home/transformations/test-remove-sidebar-upgrade-for-pro-users.php:7]:
```suggestion
use PHPUnit\Framework\TestCase as PHPUnit_TestCase;

class Test_Remove_Sidebar_Upgrade_For_Pro_Users extends PHPUnit_TestCase {
```

Elementor_Test_Base loads a bunch of stuff you don't need. Please check if the plain PHPUnit_TestCase works ok?

Comment (hein-obox) [reply tests/phpunit/elementor/modules/home/transformations/test-remove-sidebar-upgrade-for-pro-users.php]: Updated. Including Tzvi's comment.

**Review by louiswol94 — DISMISSED** [review]

**Review by louiswol94 — APPROVED** [review]

**Pull request merged**